uwblueprint/focus-on-nature
# feat: camper selection frontend

## Notion ticket link

[Cancellation-Refund-Camper-Selection-Page](https://www.notion.so/uwblueprintexecs/Cancellation-Refund-Camper-Selection-Page-All-states-9740991233214020bea547b8ae12294e)



## Implementation description
* Implemented the Camper Refund/Cancellation page
* Interacts with backend to pull data with respect to the `refundCode` in the URL and displays the available refund options
**Quick Note:** PR is not finished, still need PM or designer opinion on grouping sessions per name.


## Steps to test
1. Run the app locally and go to `localhost:3000/refund/<any-string>`. The string after refund can be anything for now, the backend will accept any string and will return the same test data
2. Check the validity of the states in each card
3. Check that the cost and numbers are correct
4. Check mobile screen sizes and verify it's what we want according to the figma


## What should reviewers focus on?
* Any styling requests?
* Double check the coding style/design to see if it's reusable/robust enough. I occasionally get errors due to the response from our Backend not being the format our frontend code expects so some functions fail
* **_As per the quick note above, would live an external opinion on grouping sessions under the same name. I feel some more decisions need to be made about this, like: "should all sessions under the same name be refunded all at once or not at all? Or have the option to individually be refunded?", "Do we need to change the formating of camper refund cards?" etc._**


## Checklist
